packages/fwfh_cached_network_image/lib/src/cached_network_image_factory.dart

```diff
@@ -5,6 +5,28 @@ import 'package:flutter_widget_from_html_core/flutter_widget_from_html_core.dart
 /// A mixin that can render IMG with `cached_network_image` plugin.
 mixin CachedNetworkImageFactory on WidgetFactory {
   @override
-  ImageProvider? imageProviderFromNetwork(String url) =>
-      url.isNotEmpty ? CachedNetworkImageProvider(url) : null;
+  Widget? buildImageWidget(BuildMetadata meta, ImageSource src) {
+    final url = src.url;
+
+    if (url.isEmpty) return null;
+
+    // Use default image loading for assets, data images and files
+    if (url.startsWith('asset:') ||
+        url.startsWith('data:image/') ||
+        url.startsWith('file:')) {
+      return super.buildImageWidget(meta, src);
+    }
+
+    return CachedNetworkImage(
+      imageUrl: url,
+      fit: BoxFit.fill,
+      placeholder: (context, url) => imageLoadingBuilder(
+        context,
+        Container(),
+        ImageChunkEvent(cumulativeBytesLoaded: 1, expectedTotalBytes: 1),
+        src,
+      ),
+      errorWidget: (_1, _2, _3) => imageErrorBuilder(_1, _3, null, src),
+    );
```

Comment:
I'm thinking about 2 changes:

1. Check for network protocol instead of other protocols. Doing this simplify things a bit.
2. Use `progressIndicatorBuilder` instead of `placeholder`. There is a performance overhead because the builder is called more often than the placeholder but it's more consistent with the `Image.loadingBuilder` (we have the real values for downloaded / total bytes)

What do you think?

```suggestion
    if (url.startsWith(RegExp('https?://')) == true) {
      return CachedNetworkImage(
        errorWidget: (context, _, error) =>
            imageErrorBuilder(context, error, null, src),
        fit: BoxFit.fill,
        imageUrl: url,
        progressIndicatorBuilder: (context, _, progress) => imageLoadingBuilder(
            context,
            const SizedBox.shrink(),
            ImageChunkEvent(
              cumulativeBytesLoaded: progress.downloaded,
              expectedTotalBytes: progress.totalSize,
            ),
            src,
          ),
      );
    }

    return super.buildImageWidget(meta, src);
```

Reply:
I'm on vacation right now, so I can't try it out. However, the changes look very nice and sound reasonable to me :)

Reply:
If you are OK with the suggested change, I can make the commit on your branch then merge this PR today. Let me know.

Reply:
That's ok for me :)